pandas/tests/base/test_ops.py

```diff
@@ -137,227 +137,233 @@ def setup_method(self, method):
         self.is_valid_objs = self.objs
         self.not_valid_objs = []
 
-    def test_none_comparison(self):
+    def test_none_comparison(self, series_with_differing_index):
+        o = series_with_differing_index
+        if isinstance(o.index, IntervalIndex):
+            pytest.skip("IntervalIndex is immutable")
```

Comment:
what does immutability have to do with this?  all indexes should be immutable

Reply:
I'll have to check. I think the IntervalIndex doesn't implement `.loc` or something like that.

Reply:
just a hunch, id imagine its like the datetime64 case where it raises for inequalities (in which case it can be added to the special case for datetime64 below)

Reply:
No, it's failing on line 149 `o[0] = np.nan` already. The assignment fails with
```
AttributeError: 'pandas._libs.interval.IntervalTree' object has no attribute 'get_loc'
```

I added a comment and adjusted the skip message. Please let me know if you have a better idea